assets/js/admin.js

```diff
@@ -124,14 +119,37 @@ jQuery(document).ready(function($) {
     var $deleteCacheButton = $( '#' + deleteCacheId );
     $deleteCacheButton.click( function(e) {
         e.preventDefault();
-        $deleteCacheButton.prop( 'disabled', true ).val( wpa0.clear_cache_working );
+        $deleteCacheButton.prop( 'disabled', true ).text( wpa0.ajax_working );
         var postData = {
             'action': deleteCacheId,
             '_ajax_nonce': wpa0.clear_cache_nonce
         };
 
         $.post(wpa0.ajax_url, postData, function() {
-            $deleteCacheButton.prop( 'disabled', false ).val( wpa0.clear_cache_done );
+            $deleteCacheButton.prop( 'disabled', false ).text( wpa0.ajax_done );
+        }, 'json');
+    } );
+
+    /*
+    Generate new migration token button on Advanced settings page
+     */
+    var rotateTokenId = 'auth0_rotate_migration_token';
+    var $rotateTokenButton = $( '#' + rotateTokenId );
+    $rotateTokenButton.click( function(e) {
+        e.preventDefault();
+
+        if (cancelAction($rotateTokenButton) ) {
+            return;
+        }
+
+        $rotateTokenButton.prop( 'disabled', true ).text( wpa0.ajax_working );
+        var postData = {
+            'action': rotateTokenId,
+            '_ajax_nonce': wpa0.rotate_token_nonce
+        };
+        $.post(wpa0.ajax_url, postData, function() {
+            $( '#auth0_migration_token' ).text(wpa0.refresh_prompt);
+            $rotateTokenButton.remove();
```

Comment:
It's kind of odd for buttons to disappear on UIs.  Is there a problem with leaving it there in case something went wrong?  Or do they really need to refresh the page?

Reply:
The function is success-only and doesn't need any data, just a 200 code. Putting the new token in the code block would be nice but would require sending the new token unnecessarily. I can't refresh the page for the user in case there are other changes that have not been saved.